Zend/zend_execute.c

```diff
@@ -4066,6 +4075,12 @@ ZEND_API void ZEND_FASTCALL zend_free_compiled_variables(zend_execute_data *exec
 		} \
 	} while (0)
 
+#define ZEND_VM_FCALL_INTERRUPT_CHECK(call) do { \
+		if (UNEXPECTED(zend_atomic_bool_exchange_ex(&EG(vm_interrupt), false))) { \
```

Comment:
Is this always faster than separate load/stores, especially when the load result is false and no store is necessary?

Reply:
It's hard to determine sometimes. It's tied to CPU and cache architectures. But it seems plausible that:

```c
if (UNEXPECTED(zend_atomic_bool_load_ex(&EG(vm_interrupt)))) {
    zend_atomic_bool_store_ex(&EG(vm_interrupt), false);
    zend_fcall_interrupt(call);
}
```

Would be faster on the hot path than this:

```c
if (UNEXPECTED(zend_atomic_bool_exchange_ex(&EG(vm_interrupt), false))) {
    zend_fcall_interrupt(call);
}
```

I don't care to quibble or test, so I made the change. It also means that `zend_fcall_interrupt` can be used by the JIT instead of having a separate `zend_jit_fcall_interrupt` and we can move the store into `zend_fcall_interrupt`.